src/llm/language_model/legacy/legacy_executor.cpp

```diff
@@ -0,0 +1,87 @@
+//*****************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "legacy_executor.hpp"
+#include "servable.hpp"
+
+namespace ovms {
+LegacyExecutor::LegacyExecutor(std::shared_ptr<ov::genai::LLMPipeline> pipe) {
```

Comment:
What is LegacyExecutor? Is it single threaded text-generation for non-continuous batching non-visual pipelines? The name states that it is something old we always had and we will be getting rid of it, which is wrong... or i dont understand something